### PR TITLE
496 minor typos in docs

### LINKED ISSUE #496: Minor Typos mainly
## Subject of the documentation
Minor typos as I read through some docs.

What are you trying to do and which part of the code is hard to understand or incorrect
or missing? Please provide a link to the readthedocs page and code if available.

I'm just pretending Im a complete beginner.

If the documentation request is approved, would you be willing to submit a PR? _(Help
can be provided if you need assistance submitting a PR)_

Sure.

# Suggestions:

* "Intensity based loss is modality-independent" sounds wrong or needs further clarification, as for example, SSD won't work on MR-CT registration. So, choice of loss function is dependent on modality.
* "measures the deformation to regularize the transformation." sounds wierd. I'd say "measures the amount of deformation, such that too severe a deformation can be penalised', as beginners may not understand that much about regularization.
* If I was a newcomer, I'd probably be lazy, and not read learn2reg. Then, by the time I had finished the first Tutorial section, I am still wondering ... "well, what sort of network is DeepReg implementing?". I'm just reading straight through the docs, and haven't seen a description of the main network.
* For adding custom loss functions, it seems odd to modify core functionality in ```dissimilarity_fn```. This is the sort of thing that should be solved with dependency injection, using a Strategy pattern, rather then editing an elif.
* Looking at ```build_backbone```, again, we are having to edit core DeepReg code. This could be improved by using an Factory pattern. 
* Both these suggestions of a StrategyPattern or Factory pattern are optional. Its more relevant for compiled code which needs to be extensible. Might be overkill for a Python/Research project. 
* The demo page for "Paired CT Lung registration", doesn't give any examples or pictures. So, its not really saying any more than the instructions on how to run the command line apps. While there is a link to source code, there is no explanation of it. 
* Same comment for several demos. Needs pictures.

### PR DESCRIPTION
**Pull request by agrimwood**
# Description

Modified tutorial text for clarification in: 1) intensity based losses and 2) deformation losses

Fixes #496 

## Type of change

What types of changes does your code introduce to DeepReg? _Put an `x` in the boxes that
apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Documentation Update (fix or improvement on the documentation)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR.
If you're unsure about any of them, don't hesitate to ask. We're here to help! This is
simply a reminder of what we are going to look for before merging your code._

- [x] I have
      [installed pre-commit](https://deepreg.readthedocs.io/en/latest/contributing/setup.html)
      using `pre-commit install` and formatted all changed files. If you are not
      certain, run `pre-commit run --all-files`.
- [x] My commits' message styles matches
      [our requested structure](https://deepreg.readthedocs.io/en/latest/contributing/commit.html),
      e.g. `Issue #<issue number>: detailed message`.
- [ ] I have updated the
      [change log file](https://github.com/DeepRegNet/DeepReg/blob/main/CHANGELOG.md)
      regarding my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by mathpluscode — APPROVED** [review]

**Comment by agrimwood** [comment]
Hi @MattClarkson , @mathpluscode , sorry my version control is a little chaotic. I have assigned you to review these very minor changes in the tutorial docs. Please feel free to suggest other reviewers if you don't have time.

File: docs/source/tutorial/registration.md [hunk 1]
```diff
@@ -74,7 +78,9 @@ The common loss function is Dice loss, Jacard and average cross-entropy over all
 
 ### Deformation loss
 
-This type of loss measures the deformation to regularize the transformation.
+This type of loss measures the amount of deformation and penalises severe deformations.
+Penalising severe deformations helps to regularise the transformation between fixed and
```
Comment (mathpluscode) [inline docs/source/tutorial/registration.md:82]: penalize or penalise lol... fine I will let it go as most of us use British english

File: docs/source/tutorial/registration.md [hunk 2]
```diff
@@ -57,9 +57,13 @@ types of losses:
...
 
 This type of loss measures the dissimilarity of the fixed image and warped moving image,
-which is adapted from the classical image registration methods. Intensity based loss is
-modality-independent and similar to many other well-studied computer vision and medical
-imaging tasks, such as image segmentation.
+which is an adaptation from classical image registration methods. Intensity based losses
+can perform poorly on multi-modality registrations (for example, SSD loss in CT-MRI
+registration). Generally, intensity based losses work best when there is an inherent
+consistency in appearance between moving and fixed images, which is more common in
+single-modality registration. Intensity based losses can be used with any modality and
```
Comment (YipengHu) [inline docs/source/tutorial/registration.md:64]: Good to make it clearer and specific. But as is would make people claiming mutual information works between different MR sequences (modalities) not very happy :) suggest to add: ... although some loss formulation such as mutual information may be effective between specific pairs of image modalities, such as some MR sequences. (or something like that)

File: docs/source/tutorial/registration.md [hunk 3]
```diff
@@ -57,9 +57,13 @@ types of losses:
...
+they complement other well-studied computer vision and medical imaging tasks, such as
```
Comment (YipengHu) [inline docs/source/tutorial/registration.md:65]: i feel after specific examples, "they complement other well-studied computer vision and medical imaging tasks..." can be removed.

File: docs/source/tutorial/registration.md [hunk 4]
```diff
@@ -74,7 +78,9 @@ The common loss function is Dice loss, Jacard and average cross-entropy over all
 
 ### Deformation loss
 
-This type of loss measures the deformation to regularize the transformation.
+This type of loss measures the amount of deformation and penalises severe deformations.
```
Comment (YipengHu) [inline docs/source/tutorial/registration.md:81]: I would also be more specific here: "penalise the non-smooth deformation", as in theory some deformation, e.g. dT/d(x,y,z)=0 will not be penalised.

**Review by YipengHu — CHANGES_REQUESTED** [review]

**Comment by mathpluscode** [comment]
Hi @agrimwood could you do some changes acc to Yipeng's comments?

After this PR and the reame one, @zacbaum I think we can do a release!

File: docs/source/tutorial/registration.md (hunk 4 above)
Comment (agrimwood) [reply docs/source/tutorial/registration.md]: Changed to: "This type of loss measures the amount of deformation in an image and penalises
non-smooth deformations. Penalising sudden or discontinuous deformations helps to
regularise the transformation between fixed and moving images."

File: docs/source/tutorial/registration.md (hunk 3 above)
Comment (agrimwood) [reply docs/source/tutorial/registration.md]: see previous change

File: docs/source/tutorial/registration.md (hunk 2 above)
Comment (agrimwood) [reply docs/source/tutorial/registration.md]: Changed paragrah to: "Intensity based losses measure the dissimilarity between a fixed image and a warped
moving image, which is an adaptation from classical image registration methods. These
losses can perform poorly on multi-modality registrations (e.g. SSD loss in CT-MRI
registration), although certain multi-modality tasks, such as registration between
different MRI sequences are handled well by MI. Generally, intensity based losses work
best when there is an inherent consistency in appearance between moving and fixed
images, which is more common in single-modality registration."

**Review by YipengHu — APPROVED** [review]

**Pull request merged**